src/core/execute.c

```diff
@@ -2953,11 +2953,10 @@ static int exec_child(
                 }
         }
 
-        if (context->nice_set)
-                if (setpriority(PRIO_PROCESS, 0, context->nice) < 0) {
-                        *exit_status = EXIT_NICE;
-                        return log_unit_error_errno(unit, errno, "Failed to set up process scheduling priority (nice level): %m");
-                }
+        if (context->nice_set) {
+                if (setpriority_closest(context->nice) < 0)
+                        log_unit_debug_errno(unit, errno, "Failed to set up process scheduling priority (nice level): %m");
```

Comment:
also, i think we should still exit on failure here? I mean setrlimit_closest() will already suppress the error cases you care about, hence in this outer handling things can remain fatal, no?

Reply:
yes